components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -362,10 +391,21 @@ void SetFingerprintingControlType(HostContentSettingsMap* map,
       primary_pattern, secondary_pattern,
       ContentSettingsType::BRAVE_FINGERPRINTING_V2, content_setting);
 
-  RecordShieldsSettingChanged(local_state);
-  if (url.is_empty()) {
-    // If global setting changed, report to P3A
-    RecordShieldsFingerprintSetting(type);
+  if (!map->IsOffTheRecord()) {
```

Comment:
It is important that we ignore any domain-specific changes made in the guest/incognito profile.